src/Executables/FindHorizons/FindHorizons.hpp

```diff
@@ -0,0 +1,325 @@
+  static void apply(const gsl::not_null<Variables<DestTagList>*> target_vars,
+                    const Variables<SrcTagList>& src_vars,
+                    const Mesh<Dim>& mesh) {
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<Dim, Frame::Inertial, DataVector>>(
+            src_vars);
+    const auto& ext_curvature =
+        get<gr::Tags::ExtrinsicCurvature<Dim, Frame::Inertial, DataVector>>(
+            src_vars);
+    const auto& inv_jacobian =
+        get<domain::Tags::InverseJacobian<Dim, Frame::ElementLogical,
+                                          Frame::Inertial>>(src_vars);
+    get<gr::Tags::SpatialMetric<Dim, Frame::Inertial, DataVector>>(
+        *target_vars) = spatial_metric;
+    get<gr::Tags::ExtrinsicCurvature<Dim, Frame::Inertial, DataVector>>(
+        *target_vars) = ext_curvature;
+    auto& inv_spatial_metric =
+        get<gr::Tags::InverseSpatialMetric<Dim, Frame::Inertial, DataVector>>(
+            *target_vars);
+    Scalar<DataVector> unused_det{mesh.number_of_grid_points()};
+    determinant_and_inverse(make_not_null(&unused_det),
+                            make_not_null(&inv_spatial_metric), spatial_metric);
+    const auto deriv_spatial_metric =
+        ::partial_derivative(spatial_metric, mesh, inv_jacobian);
+    auto& spatial_christoffel_second_kind =
+        get<gr::Tags::SpatialChristoffelSecondKind<Dim, Frame::Inertial,
+                                                   DataVector>>(*target_vars);
+    gr::christoffel_second_kind(make_not_null(&spatial_christoffel_second_kind),
+                                deriv_spatial_metric, inv_spatial_metric);
+    const auto deriv_spatial_christoffel_second_kind = ::partial_derivative(
+        spatial_christoffel_second_kind, mesh, inv_jacobian);
+    auto& spatial_ricci =
+        get<gr::Tags::SpatialRicci<Dim, Frame::Inertial>>(*target_vars);
+    gr::ricci_tensor(make_not_null(&spatial_ricci),
+                     spatial_christoffel_second_kind,
+                     deriv_spatial_christoffel_second_kind);
+  }
+};
+
+template <size_t Dim, ah::ObjectLabel Label>
+struct ApparentHorizon
+    : tt::ConformsTo<intrp::protocols::InterpolationTargetTag> {
+  static std::string name() { return "Ah" + ah::name(Label); }
+  using temporal_id = ::Tags::Time;
+  using compute_target_points =
+      intrp::TargetPoints::ApparentHorizon<ApparentHorizon, Frame::Inertial>;
+  using post_interpolation_callback =
+      intrp::callbacks::FindApparentHorizon<ApparentHorizon, Frame::Inertial>;
+  using horizon_find_failure_callback =
+      intrp::callbacks::ErrorOnFailedApparentHorizon;
+  using post_horizon_find_callbacks = tmpl::list<
+      intrp::callbacks::ObserveTimeSeriesOnSurface<
+          ::ah::tags_for_observing<Frame::Inertial>, ApparentHorizon>,
+      intrp::callbacks::ObserveSurfaceData<::ah::surface_tags_for_observing,
+                                           ApparentHorizon, ::Frame::Inertial>>;
```

Comment:
Could you also add `ah::callbacks::ObserveCenters`? That info is also useful.

Reply:
That requires both grid and inertial frame tags in the DataBox. Currently only the inertial frame is used.

Reply:
Currently, all the tags in `::ah::tags_for_observing` and `ah::surface_tags_for_observing` are either scalars or doubles which don't depend on what frame you find the horizon in. Also we output the inertial coordinates regardless. So the output we'd get is the same regardless of what frame we use.

Would you be ok switching the horizon find to happen in the grid frame so you could also observe the centers? Diagnostically, the centers are pretty important so I'd like to have that info available. Hopefully it's as simple as just changing the `Frame::Inertial` tag in this struct to `Frame::Grid`.

Reply:
I tried this here: https://github.com/nilsvu/spectre/tree/find_horizons_exec_gridframe. However, der interpolation framework can't deal with the grid-to-inertial transformation if the domain is time-independent, because no grid frame will ever be constructed. I'm sure we can find workarounds for that (e.g. set up an identity Jacobian here: https://github.com/sxs-collaboration/spectre/blob/develop/src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp#L669), but doing that in a clean way goes a bit over the scope of this PR I think. So keep this in the inertial frame for now and adapt to the moving BBH domain later (alongside loading in the FoT parameters etc) sound ok?